cache/async_cache.go

```diff
@@ -29,22 +31,45 @@ func (c *AsyncCache) Close() error {
 	return nil
 }
 
-func (c *AsyncCache) AwaitForConcurrentTransaction(key *Key) bool {
-	startTime := time.Now()
+type TransactionResult struct {
+	ElapsedTime time.Duration
+	State       TransactionState
+}
 
+func (c *AsyncCache) AwaitForConcurrentTransaction(key *Key) (TransactionResult, error) {
+	startTime := time.Now()
+	var seenState TransactionState
```

Comment:
It might be more readable to initialize seenState at transactionCreated because without knowing TransactionState you can't understand this part